commit 470b5bdd8fd29d6b35e3a80f9a57bdd4b2438200
Author: Tobias Stoeckmann <tobias@stoeckmann.org>
Date:   Fri Jan 19 10:09:00 2024 +0100

    pam_unix: do not warn if password aging is disabled
    
    Later checks will print a warning if daysleft is 0. If password
    aging is disabled, leave daysleft at -1.
    
    Resolves: https://github.com/linux-pam/linux-pam/issues/743
    Fixes: 9ebc14085a3b ("pam_unix: allow disabled password aging")
    Signed-off-by: Tobias Stoeckmann <tobias@stoeckmann.org>

diff --git a/modules/pam_unix/passverify.c b/modules/pam_unix/passverify.c
index 5c4f862e..1bc98fa2 100644
--- a/modules/pam_unix/passverify.c
+++ b/modules/pam_unix/passverify.c
@@ -314,7 +314,6 @@ PAMH_ARG_DECL(int check_shadow_expiry,
 	}
 	if (spent->sp_lstchg < 0) {
 		D(("password aging disabled"));
-		*daysleft = 0;
 		return PAM_SUCCESS;
 	}
 	if (curdays < spent->sp_lstchg) {
